### ISSUE TITLE
indentation_width warning when preprocessor statements are used

### ISSUE THREAD

**Issue by phranck** [realm/SwiftLint#4121]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

It was fixed in #4066, unfortunately not completely. Please have a look at: https://github.com/realm/SwiftLint/issues/4030#issuecomment-1211205958.

**Comment by SimplyDanny** [comment]
Works on my machine ... 😄

Are you sure the new option is set to `false` and you are using one of the release candidates for version 0.49.0? The fix is not contained in version 0.48.0 or earlier.

**Comment by SwiftNativeDeveloper** [comment]
This does not work on my machine and my colleagues. We have both have 0.50.3 installed.

**Comment by SimplyDanny** [comment]
@SwiftNativeDeveloper, @phranck: Please try `swiftlint lint --no-cache` and let me know whether the warnings disappear.

**Comment by SwiftNativeDeveloper** [comment]
> @SwiftNativeDeveloper, @phranck: Please try `swiftlint lint --no-cache` and let me know whether the warnings disappear.

My colleague cleaned derived data and restarted Xcode and that worked for him. Next time we have stuck settings I'll share the --no-cache flag. Thanks!

**Comment by phranck** [comment]
@SimplyDanny 
Unfortunately, this rule still doesn't work.  
I have a "Run Script Phase" containing the following:

```
export PATH="$PATH:/opt/homebrew/bin"
if which swiftlint > /dev/null; then
  swiftlint lint --no-cache
else
  echo "warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint"
fi
```

I have opt in the rule `indentation_width` with this setting:

```
indentation_width:
  include_compiler_directives: false
```

and I still get the warning:

<img width="1500" alt="image" src="https://user-images.githubusercontent.com/871198/215301188-a2fe2dea-b5c9-454c-8d19-661fc6d001a1.png">

**Comment by SimplyDanny** [comment]
Does it work on the plain command line without any Xcode and build scripts involved?

**Comment by phranck** [comment]
Yes, indeed. There it does. The warnings from screenshot above are from `MainView.swift`.

<img width="952" alt="image" src="https://user-images.githubusercontent.com/871198/215325947-9f6ccad2-40bc-434e-88f9-c14969145752.png">

**Comment by SimplyDanny** [comment]
Are you sure the configuration file is taken into account when running SwiftLint as a build step? Maybe try with an explicit `--config` option.

**Comment by phranck** [comment]
This is strange. The `MyProject.xcodeproj` and the `.swiftlint.yml` are in the same folder. Running `swiftlint --no-cache` from within this folder everyhting is fine.

Calling the linter in the build phase script with `swiftlint --no-cache --config ${PROJECT_DIR}/.swiftlint.yml` it doesn't work. `${PROJECT_DIR}` points exactly to the folder where `MyProject.xcodeproj` is and `swiftlint` is being called from this folder while the build phase script is running. 🤔

**Comment by phranck** [comment]
Am I the only one still seeing the warnings? 🤔
My current Runs Script Phase looks like this, and still warnings...

```shell
export PATH="$PATH:/opt/homebrew/bin"
if which swiftlint > /dev/null; then
  swiftlint --no-cache --config ${PROJECT_DIR}/.swiftlint.yml
else
  echo "warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint"
fi
```
